src/utils/format.ts

```diff
@@ -6,18 +6,25 @@ import { TEZ, TEZ_DECIMALS } from "./tezos";
 export const truncate = (name: string, len: number) =>
   name.length > len ? name.slice(0, len - 3) + "..." : name;
 
-export const tezToMutez = (tez: string): BigNumber => format("tz", "mutez", tez) as BigNumber;
+export const tezToMutez = (tez: string): string | number | BigNumber => format("tz", "mutez", tez);
```

Comment:
if tez is a valid number then it'll be always BigNumber. I think it's fine to leave it as it was. for safety you could throw here if you want https://github.com/ecadlabs/taquito/blob/master/packages/taquito-utils/src/format.ts#L27